lxd/storage/s3/miniod/admin.go

```diff
@@ -0,0 +1,226 @@
+package miniod
+
+import (
+	"archive/zip"
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/canonical/lxd/shared"
+	"github.com/canonical/lxd/shared/api"
+)
+
+type minioAdmin struct {
+	alias string
+}
+
+// AccountInfo is the response body of the info service account call.
+type AccountInfo struct {
+	ParentUser    string          `json:"parentUser"`
+	AccountStatus string          `json:"accountStatus"`
+	ImpliedPolicy bool            `json:"impliedPolicy"`
+	Policy        json.RawMessage `json:"policy"`
+}
+
+// Credentials holds access and secret keys.
+type Credentials struct {
+	AccessKey    string    `xml:"AccessKeyId"     json:"accessKey"`
+	SecretKey    string    `xml:"SecretAccessKey" json:"secretKey"`
+	SessionToken string    `xml:"SessionToken"    json:"sessionToken"`
+	Expiration   time.Time `xml:"Expiration"      json:"expiration"`
+}
+
+// ServiceAccountArgs is the request options for adding or modifying a service account.
+type ServiceAccountArgs struct {
+	Policy    json.RawMessage `json:"policy"` // Parsed value from iam/policy.Parse()
+	AccessKey string          `json:"accessKey"`
+	SecretKey string          `json:"secretKey"`
+}
+
+// configJSON represents the JSON payload containing the configuration dump.
+type configJSON struct {
+	Status string `json:"status"`
+	Value  []byte `json:"value"`
+}
+
+// NewAdminClient returns a new AdminClient, and assigns the given credentials to an alias, and returns an `mc` client for that alias.
+func NewAdminClient(url string, username string, password string) (*minioAdmin, error) {
+	_, err := shared.RunCommand("mc", "alias", "set", "--insecure", "lxd-minio", api.NewURL().Scheme("http").Host(url).String(), username, password)
```

Comment:
Could we have a constant for "lxd-minio" as you've used it in 2 places here, do they represent the same thing?